dcpy/extract/ingest.py

```diff
@@ -84,10 +90,92 @@ def extract_and_archive_raw_dataset(dataset: str, version: str | None):
     return config
 
 
-def transform_to_parquet(config: recipes.ExtractConfig):
-    """Given config of archived raw dataset, transform it to parquet"""
-    ### idea is that this will dump to PARQUET_PATH - other functions will assume parquet file is there as well
-    raise NotImplemented
+def transform_to_parquet(
+    config: recipes.ExtractConfig, local_data_path: Path | None = None
+):
+    """
+    Transforms raw data into a parquet file format and saves it locally.
+
+    This function first checks for the presence of raw data locally at the specified `local_data_path`.
+    If the path not provided, it is downloaded from an S3 bucket using the `config` parameter.
+    The raw data is then read into a GeoDataFrame and saved as a parquet file.
+
+    The transformation process varies depending on the format of the raw data, which can be in .shp, .gdb,
+    or .csv format. For csv files, if geometry is present, it is converted into a GeoSeries before creating
+    the GeoDataFrame.
+
+    Parameters:
+        config (recipes.ExtractConfig): Config object containing geometry info.
+        local_data_path (Path, optional): Path to the local data file. If not provided, data is pulled from S3 bucket.
+
+    Raises:
+        AssertionError: If `local_data_path` is provided but does not point to a valid file.
+    """
+
+    # create new dir for raw data and output parquet file
+    if TMP_DIR.is_dir():
+        shutil.rmtree(TMP_DIR)
+    TMP_DIR.mkdir()
+
+    if local_data_path:
+        # TODO: fix logic for cases when it's a directory, not a file
+        assert local_data_path.is_file(), "Local path should be a valid file"
```

Comment:
if this raises an error when `local_data_path` is a directory, that seems like a good thing

Reply:
I was thinking about a zipped shape file... But it's probably considered as a file, not a folder

Reply:
Update: `gdb` format is considered a directory. My test didn't catch the assert statement because I misspelled geodatabase as geodabase 🥲 Socrate moment